extension/test/integration/sessions-request.handler.spec.js

```diff
@@ -0,0 +1,111 @@
+import { expect } from 'chai'
+import _ from 'lodash'
+import ctpClientBuilder from '../../src/ctp.js'
+import c from '../../src/config/constants.js'
+import config from '../../src/config/config.js'
+import utils from '../../src/utils.js'
+
+describe('::create-session-request::', () => {
+  let packageJson
+  let ctpClient
+  const [commercetoolsProjectKey] = config.getAllCtpProjectKeys()
+  const [adyenMerchantAccount] = config.getAllAdyenMerchantAccounts()
+
+  before(async () => {
+    packageJson = await utils.readAndParseJsonFile('package.json')
+    const ctpConfig = config.getCtpConfig(commercetoolsProjectKey)
+    ctpClient = await ctpClientBuilder.get(ctpConfig)
+  })
+
+  it(
+    'given a payment ' +
+      'when createSessionRequest custom field is set and custom field createSessionResponse is not set ' +
+      'then should set custom field createSessionResponse ' +
+      'and interface interaction with type createSession',
+    async () => {
+      const createSessionRequestDraft = {
+        countryCode: 'DE',
+        reference: 'UNIQUE_PAYMENT_REFERENCE',
+        returnUrl: '/',
+        amount: {
+          currency: 'EUR',
+          value: 1000,
+        },
+      }
+      const paymentDraft = {
+        amountPlanned: {
+          currencyCode: 'EUR',
+          centAmount: 1000,
+        },
+        paymentMethodInfo: {
+          paymentInterface: c.CTP_ADYEN_INTEGRATION,
+        },
+        custom: {
+          type: {
+            typeId: 'type',
+            key: c.CTP_PAYMENT_CUSTOM_TYPE_KEY,
+          },
+          fields: {
+            createSessionRequest: JSON.stringify(createSessionRequestDraft),
+            commercetoolsProjectKey,
+            adyenMerchantAccount,
+          },
+        },
+      }
+
+      const { statusCode, body: payment } = await ctpClient.create(
+        ctpClient.builder.payments,
+        paymentDraft
+      )
+
+      const createSessionRequestExtended = _.cloneDeep(
+        createSessionRequestDraft
+      )
+      createSessionRequestExtended.applicationInfo = {
+        merchantApplication: {
+          name: packageJson.name,
+          version: packageJson.version,
+        },
+        externalPlatform: {
+          name: 'commercetools',
+          integrator: packageJson.author.name,
+        },
+      }
+      expect(statusCode).to.equal(201)
+
+      const { createSessionRequest, createSessionResponse } =
+        payment.custom.fields
+      expect(createSessionRequest).to.be.deep.equal(
+        JSON.stringify(createSessionRequestDraft)
+      )
+
```

Comment:
Please add assertion to make sure the payment key is equal to the reference inside create session request

Reply:
[52b0e90](https://github.com/commercetools/commercetools-adyen-integration/pull/1051/commits/52b0e9088c1b3c7c856d7d5a0d768bb42814fd97)